assets/scss/base/_links.scss

```diff
@@ -35,3 +35,11 @@ a[rel="external"].external-link-19-bold-no-hover {
 .link--no-underline {
   text-decoration: none;
 }
+
+.link--hover-underline {
+  text-decoration: none;
+  &:hover, 
+  &:focus { 
+    text-decoration: underline;  
+  }
+}
```

Comment:
according to the name this should be:
```
.link--hover-underline {
    &:hover, 
    &:focus { 
        text-decoration: underline;  
    }
}
```

and then use `link--no-underline` on the link also?